ml-agents/mlagents/trainers/torch/model_serialization.py

```diff
@@ -68,18 +68,38 @@ def __init__(self, policy):
             for obs_spec in self.policy.behavior_spec.observation_specs
             if len(obs_spec.shape) == 3
         ]
+
+        dummy_var_len_obs = [
+            torch.zeros(batch_dim + [obs_spec.shape[0], obs_spec.shape[1]])
+            for obs_spec in self.policy.behavior_spec.observation_specs
+            if len(obs_spec.shape) == 2
+        ]
+
         dummy_masks = torch.ones(
             batch_dim + [sum(self.policy.behavior_spec.action_spec.discrete_branches)]
         )
         dummy_memories = torch.zeros(
             batch_dim + seq_len_dim + [self.policy.export_memory_size]
         )
 
-        self.dummy_input = (dummy_vec_obs, dummy_vis_obs, dummy_masks, dummy_memories)
+        self.dummy_input = (
+            dummy_vec_obs,
+            dummy_vis_obs,
+            dummy_var_len_obs,
+            dummy_masks,
+            dummy_memories,
+        )
 
         self.input_names = (
             ["vector_observation"]
             + [f"visual_observation_{i}" for i in range(num_vis_obs)]
+            + [
+                f"obs_{i}"
+                for i, sens_spec in enumerate(
```

Comment:
nit: obs_spec (and similar earlier in the method)

Reply:
I think this would clearer as
```
self.input_names = ["vector_observation"]
for i in range(num_vis_obs):
  self.input_names.append("visual_observation_{i}")
for i, obs_spec in enumerate(self.policy.behavior_spec.observation_specs):
  if len(obs_spec.shape) == 2:
    self.input_names.append("obs_{i}")
self.input_names += ["action_masks", "memories"]
```
You can maybe also avoid some line breaks with `observation_specs = self.policy.behavior_spec.observation_specs` earlier in the method.